### PR TITLE
Add OpenCensus Tracing support

### PR DESCRIPTION
**Pull request by cschuet**
Add OpenCensus Tracing support

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,3 @@
+BasedOnStyle: Google
```
Comment (gaschler) [inline .clang-format]: separate PR would be nice

File: WORKSPACE [hunk 2]
```diff
@@ -21,3 +21,9 @@ repositories()
 load("@com_github_grpc_grpc//bazel:grpc_deps.bzl", "grpc_deps")
 
 grpc_deps()
+
+load("@com_github_jupp0r_prometheus_cpp//:repositories.bzl", "load_prometheus_client_model", "load_civetweb")
```
Comment (gaschler) [inline WORKSPACE]: do you need load_prometheus_client_model?
(I cannot see where it is used.)

File: async_grpc/BUILD.bazel [hunk 3]
```diff
@@ -61,11 +61,14 @@ cc_library(
     ]),
     copts = ["-Wno-sign-compare"],
     includes = ["."],
+    defines = ["TRACING_SUPPORT"],
```
Comment (gaschler) [inline async_grpc/BUILD.bazel]: Cartographer style: rename to BUILD_TRACING(_SUPPORT)
maybe copts is enough unless this needs to be exported

File: async_grpc/rpc_handler.h [hunk 4]
```diff
@@ -21,12 +21,12 @@
 #include "async_grpc/rpc.h"
 #include "async_grpc/rpc_handler_interface.h"
 #include "async_grpc/rpc_service_method_traits.h"
-
 #include "google/protobuf/message.h"
-
 #include "glog/logging.h"
-
 #include "grpc++/grpc++.h"
+#ifdef TRACING_SUPPORT
```
Comment (gaschler) [inline async_grpc/rpc_handler.h]: In Cartographer, we use `#if USE_PROMETHEUS`.
ifdef has the disadvantage it is unclear when it is not set, and awkward to use in logic, so I slightly prefer if.

File: async_grpc/rpc_handler.h [hunk 5]
```diff
@@ -21,12 +21,12 @@
...
+#include "opencensus/trace/span.h"
```
Comment (gaschler) [inline async_grpc/rpc_handler.h]: would be good to wrap (add todo)

File: async_grpc/server.cc [hunk 6]
```diff
@@ -47,6 +55,30 @@ void Server::Builder::SetMaxSendMessageSize(int max_send_message_size) {
   options_.max_send_message_size = max_send_message_size;
 }
 
+void Server::Builder::EnableTracing() {
+#ifdef TRACING_SUPPORT
+  options_.enable_tracing = true;
+#else
+  LOG(FATAL) << "Enable tracing support by setting -DTRACING_SUPPORT first.";
```
Comment (gaschler) [inline async_grpc/server.cc]: "support by compiling with"

File: bazel/repositories.bzl [hunk 7]
```diff
@@ -15,85 +15,168 @@
...
+        strip_prefix = "benchmark-master",
+        sha256 = "4339fd994efd0cc8166a9bbb87f9e9f518b5de0bf29cf36999bd434c35ca7505",
+    )
+
+    _maybe(
+        native.new_http_archive,
+        name = "com_github_curl",
+        urls = ["https://github.com/curl/curl/archive/e2ef8d6fa11b2345e10b89db525920f2a0d5fd79.zip"],
+        strip_prefix = "curl-e2ef8d6fa11b2345e10b89db525920f2a0d5fd79",
+        build_file_content =
```
Comment (gaschler) [inline bazel/repositories.bzl]: also use build_file and write this into separate bazel/third_party file?

**Review by gaschler — APPROVED** [review]
Nice feature!

File: .clang-format (hunk 1 above)
Comment (cschuet) [reply .clang-format]: done

File: WORKSPACE (hunk 2 above)
Comment (cschuet) [reply WORKSPACE]: removed. Not sure why I thought it is needed.

File: async_grpc/BUILD.bazel (hunk 3 above)
Comment (cschuet) [reply async_grpc/BUILD.bazel]: done

File: async_grpc/rpc_handler.h (hunk 4 above)
Comment (cschuet) [reply async_grpc/rpc_handler.h]: done

File: async_grpc/rpc_handler.h (hunk 5 above)
Comment (cschuet) [reply async_grpc/rpc_handler.h]: done.

File: async_grpc/server.cc (hunk 6 above)
Comment (cschuet) [reply async_grpc/server.cc]: done.

File: bazel/repositories.bzl (hunk 7 above)
Comment (cschuet) [reply bazel/repositories.bzl]: done.

**Pull request merged**